cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -12,14 +12,24 @@
 import io.vertx.core.json.Json;
 import io.vertx.core.json.JsonObject;
 
+import java.net.ConnectException;
+import java.util.concurrent.TimeoutException;
+
 public class CruiseControlApiImpl implements CruiseControlApi {
 
     private static final boolean HTTP_CLIENT_ACTIVITY_LOGGING = false;
+    private static final int HTTP_DEFAULT_IDLE_TIMEOUT_SECONDS = 0;
```

Comment:
So by default we are not waiting at all for a slow CC reponse?

Reply:
No, we are waiting. When the `idleTimeout` is 0 (the default here) we don't set the timeout on the request, which means using the default internal timeout.

Reply:
Oh right. That is ok then.

Reply:
IMHO it's confusing for a timeout of 0 to mean "use the default". Can we use -1 instead, since it's obviously not valid at a timeout.

Reply:
Yeah that's a fair point.